model/models/threes.py

```diff
@@ -114,16 +123,21 @@ def run_model(
         def_adjustment = opponent_fg_percent_by_cluster / league_fg_percent_by_cluster
 
         #bootstrap resample from FGA data to find normal distribution fo estimated mean FGA per game
-        fga_per_game_est = [np.random.choice(fga_per_game_data, size=len(fga_per_game_data), replace=True).mean() for _ in range(bootstrap_samples)]
-        fga_per_game_est_mean = np.mean(fga_per_game_est)
-        fga_per_game_est_std = np.std(fga_per_game_est)
+        if fga_method == 'simple':
+            fga_per_game_array = fga_per_game_data['SHOT_ATTEMPTED_FLAG'].values
+            fga_per_game_est = [np.random.choice(fga_per_game_array, size=len(fga_per_game_array), replace=True).mean() for _ in range(bootstrap_samples)]
+            fga_per_game_est_mean = np.mean(fga_per_game_est)
+            fga_per_game_est_std = np.std(fga_per_game_est)
+        elif fga_method == 'ewm':
+            fga_per_game_est_mean = fga_per_game_data['SHOT_ATTEMPTED_FLAG'].ewm(span = len(fga_per_game_data)).mean().values[-1]
+            fga_per_game_est_std = fga_per_game_data['SHOT_ATTEMPTED_FLAG'].ewm(span = len(fga_per_game_data)).std().values[-1]
```

Comment:
No bootstrapping is getting done here, won't be able to compress into a list comprehension but I think we can write a for loop to do it.
eg. 
```python
bootstrapped_emas = []
for _ in range(bootstrap_samples):
    fga_boot = np.random.choice(fga_per_game_array, size=len(fga_per_game_array), replace=True)
    fga_boot_ema = pd.ewma(fga_boot, span=len(fga_boot)).mean()[-1]
    bootstrapped_emas.append(fga_boot_ema)
fga_per_game_est_mean = np.mean(bootstrapped_means)
fga_per_game_est_std = np.std(bootstrapped_means)
```

I've never actually done this before and I'd like to actually examine the distribution here in a notebook or something. Since EMA is dependent on the ordering I'm not sure this is valid, but I think it does (mentioned this point earlier)

Reply:
Actually having messed around with this in a notebook i dont think it does work. ordering does matter